award_emojis.go

```diff
@@ -0,0 +1,219 @@
+//
+// Copyright 2017, Arkbriar
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+	"time"
+)
+
+// AwardEmojiService handles communication with the emoji awards related methods
+// of the GitLab API.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/award_emoji.html
+type AwardEmojiService struct {
+	client *Client
+}
+
+// AwardEmoji represents a GitLab Award Emoji.
+//
+// GitLab API docs: https://docs.gitlab.com/ce/api/award_emoji.html
+type AwardEmoji struct {
+	ID   int    `json:"id"`
+	Name string `json:"name"`
+	User struct {
+		Name      string `json:"name"`
+		Username  string `json:"username"`
+		ID        int    `json:"id"`
+		State     string `json:"state"`
+		AvatarURL string `json:"avatar_url"`
+		WebURL    string `json:"web_url"`
+	} `json:"user"`
+	CreatedAt     time.Time `json:"created_at"`
+	UpdatedAt     time.Time `json:"updated_at"`
+	AwardableID   int       `json:"awardable_id"`
+	AwardableType string    `json:"awardable_type"`
+}
+
+// ListAwardEmoji gets a list of all award emoji on the specified resource
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ce/api/award_emoji.html#list-an-awardable-39-s-award-emoji
+func (a *AwardEmojiService) ListAwardEmoji(pid interface{}, resource string, resourceID int, options ...OptionFunc) ([]*AwardEmoji, *Response, error) {
```

Comment:
It seems like we should find a way to handle the resource type without requiring the caller to pass it as a string.

A couple options:
```
const (
    AwardMergeRequest = "merge_request"
    // The rest
)
```

Could unexport these functions and use them as a factory
```
func (a *AwardEmojiService) ListMergeRequestAwardEmoji(...) ... {
   return s.listAwardEmoji(pid, "merge_request", rid)
}
```